version: "3.8"

services:
  nifi_2_0:
    build: 
      dockerfile: ./dockerfile
      context: .
    image: renatoelho/nifi:2.0.0
    hostname: nifi_2_0
    container_name: nifi_2_0
    environment:
      - JAVA_HOME=/usr/lib/jvm/jdk-21-oracle-x64
      - PYTHONPATH=/usr/bin/python3
      - PATH=$PATH:/home/nifi/nifi-2.0.0-M2/bin
      - TZ=America/Sao_Paulo
      - TERM=xterm-256color
    ports:
      - "8080:8080"
    deploy:
      resources:
        limits:
          cpus: "2.0"
          memory: 4G
    restart: on-failure
    volumes:
      - nifi_bin:/home/nifi/nifi-2.0.0-M2/bin
      - nifi_conf:/home/nifi/nifi-2.0.0-M2/conf
      - nifi_extencions:/home/nifi/nifi-2.0.0-M2/extensions
      - nifi_python:/home/nifi/nifi-2.0.0-M2/python
    #healthcheck:
    #  test: curl -f http://app_gerafacil:8001 || exit 1
    #  interval: 60s
    #  timeout: 15s
    #  retries: 5
    networks:
      - rede-nifi-2-0

networks:
  rede-nifi-2-0:
    driver: bridge

volumes:
  nifi_bin:
  nifi_conf:
  nifi_extencions:
  nifi_python:

